generation/generate_release_please_config.sh

```diff
@@ -30,7 +30,7 @@ for path in $module_list; do
   module_snapshot_version=$(echo "${module_line}" | cut -d ":" -f3)
 
   # concatenating module name and module version
-  rp_manifest_line="${tab}\"${module_name}\": \"${module_released_version}\""
```

Comment:
I thought the release-please manifest should always have the latest released version. Wouldn't this potentially change the versions in the manifest to a snapshot version depending on when this gets run in the future?

Reply:
Regarding point #1, In the release:main branch, the versions will be like {release_version}:{release_version}.
At this point, our script will be generating non-snapshot versions. This is coherent with non-snapshot versions in manifest file.

We release, then the main will have {release_version}:{release_version}. All good till now.

Then we have snapshot PR. This PR is changing versions from `{release_version}:{release_version}` to  `{release_version}:{current_version-SNAPSHOT}` in versions.txt. Also, the manifest file will change versions to snapshot too. We merge this in. once merged, our main will have `{release_version}:{current_version-SNAPSHOT}` in versions.txt and snapshot versions in manifest.

Next we have a new library in monorepo. We run the script to generate the config again. Our config will have snapshot versions in it. The next release:main PR will change the versions from snapshot to non-snapshot in manifest file, which is what we need to release.

This is how it should ideally work, in my understanding.

Reply:
Also when I think about it, if release-please does not update the manifest file in the snapshot PR, that would be the best solution.

Reply:
Ah gotcha, I didn't know the manifest also got updated to `-SNAPSHOT`. I thought it was the source of truth for each module's last released version and only held the latest released versions.

I think I understand the flow now, but correct me if I'm wrong: 
For any new modules, the versions.txt file is `0.0.0:0.0.1-SNAPSHOT`. This would mean that if the manifest is generated again with new module, it would use {current_version} which is `0.0.1-SNAPSHOT`. So going forward, we would add in new modules *ONLY* after the release-please `-SNAPSHOT` PR has been merged in (since this would make every module have `-SNAPSHOT`). Then we can run the `release:main` PR for the actual release.

Reply:
In java-cloud-bom repository (which uses manifest-based configuration) the ".release-please-manifest.json" has snapshot versions https://github.com/googleapis/java-cloud-bom/blob/main/.release-please-manifest.json

Reply:
"So going forward, we would add in new modules ONLY after the release-please -SNAPSHOT PR has been merged in" -> Ideally, as soon as a release is done, the snapshot PR is created and should be merged in. After that, any additional work should be done for the next release.

Reply:
Yep, makes sense given the `-SNAPSHOT` flow.